### PR TITLE
Bug 2257371: external: include 255 as a valid ip range

### PR DESCRIPTION
**Pull request by subhamkrai**
**Description of your changes:**
external: include 255 as a valid ip range
    
    external script was checking ip range up to 254
    which is incorrect, the valid range is 0-255
    and if someone passes 255 it will through error like
    ```
    Out of range IP addresses
    ```
    so including 255 as a valid range.

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure)).
- [ ] **Skip Tests for Docs**: If this is only a documentation change, add the label `skip-ci` on the PR.
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/Contributing/development-flow/#submitting-a-pull-request)
- [ ] [Pending release notes](https://github.com/rook/rook/blob/master/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

File: deploy/examples/create-external-cluster-resources.py [hunk 1]
```diff
@@ -549,7 +549,7 @@ def _invalid_endpoint(self, endpoint_str):
                     f"IP address parts should be numbers: {ipv4}"
                 )
             intPart = int(eachPart)
-            if intPart < 0 or intPart > 254:
+            if intPart < 0 or intPart > 255:
```
Comment (parth-gr) [inline deploy/examples/create-external-cluster-resources.py:552]: in the unit test you can check, can you add a small test
And do we need upstream fix for this in lower version

**Review by parth-gr — CHANGES_REQUESTED** [review]

File: deploy/examples/create-external-cluster-resources.py (hunk 1 above)
Comment (subhamkrai) [reply deploy/examples/create-external-cluster-resources.py:552]: I'm not planning to add a unit test for this as the future version already has an appropriate fix. And I don't think we are doing any release for the upstream version of this older version.

**Review by parth-gr — APPROVED** [review]

**Comment by subhamkrai** [comment]
@travisn CI errors are fixed in newer versions, I don't think my changes are affecting this.

Let me know wdyt, should we go ahead with these changes or need to fix CI.

**Comment by travisn** [comment]
@subhamkrai Agreed the CI issues are not related. We can merge this when it's approved.

**Comment by subhamkrai** [comment]
I think we can go ahead with PR multiple CI fixes are not backported to 4.14 and earlier branches, For examplehttps://github.com/red-hat-storage/rook/pull/569

**Pull request merged**